pkg/operators/ebpf/ebpf.go

```diff
@@ -676,6 +686,34 @@ func (i *ebpfInstance) Start(gadgetCtx operators.GadgetContext) error {
 	}
 	i.collection = collection
 
+	if i.bpfOperator.gadgetObjs != nil {
```

Comment:
We don't need to collect program and map IDs for this gadget if the emitstats annotation wasn't set. I know it will do nothing anyway, but this operation is anyway useless in that case.

Reply:
Ahh ok, because the `emitstats` is not an annotation we add to the gadget we want to collect stats from, but to the "empty" gadget we are running to gather the collected stats. Ok, ok. Understood. Thanks for the explanation.